hypershift-operator/controllers/hostedcluster/metrics.go

```diff
@@ -92,3 +99,27 @@ func clusterVersionRolloutTime(hc *hyperv1.HostedCluster) *float64 {
 	}
 	return pointer.Float64(completionTime.Sub(hc.CreationTimestamp.Time).Seconds())
 }
+
+func RecordClusterUpgradeDuration(hc *hyperv1.HostedCluster) {
```

Comment:
So just to clarify this would record the whole history every time that is run? which I guess is fine.

Reply:
right